src/zampy/datasets/utils.py

```diff
@@ -54,3 +68,137 @@ def get_file_size(fpath: Path) -> int:
         return 0
     else:
         return fpath.stat().st_size
+
+
+def cds_request(  # noqa: PLR0913
+    product: str,
+    variables: List[str],
+    time_bounds: TimeBounds,
+    spatial_bounds: SpatialBounds,
+    path: Path,
+    overwrite: bool,
+) -> None:
+    """Download data via CDS API.
+
+    To raise a request via CDS API, the user needs to set up the
+    configuration file `.cdsapirc` following the instructions on
+    https://cds.climate.copernicus.eu/api-how-to.
+
+    Following the efficiency tips of request,
+    https://confluence.ecmwf.int/display/CKB/Climate+Data+Store+%28CDS%29+documentation
+    The downloading is organized by asking for one month of data per request.
+
+    Args:
+        product: Dataset name for retrieval via `cdsapi`.
+        variables: Zampy variable.
+        time_bounds: Zampy time bounds object.
+        spatial_bounds: Zampy spatial bounds object.
+        path: File path to which the data should be saved.
+        overwrite: If an existing file (of the same size!) should be overwritten.
+    """
+    fname = PRODUCT_FNAME[product]
+
+    with CDSAPI_CONFIG_PATH.open(encoding="utf8") as f:
+        url = f.readline().split(":", 1)[1].strip()
+        api_key = f.readline().split(":", 1)[1].strip()
+
+    c = cdsapi.Client(
+        url=url,
+        key=api_key,
+        verify=True,
+        quiet=True,
+    )
+
+    # create list of year/month pairs
+    year_month_pairs = time_bounds_to_year_month(time_bounds)
+
+    for (year, month), variable in itertools.product(year_month_pairs, variables):
+        # check existence and overwrite
+        fpath = path / f"{fname}_{variable}_{year}-{month}.nc"
+        if fpath.exists() and not overwrite:
+            print(f"File '{fpath.name}' already exists, skipping...")
+            continue
+        # raise download request
+        c.retrieve(
+            product,
+            {
+                "product_type": "reanalysis",
+                "variable": [variable],
+                "year": year,
+                "month": month,
+                "day": [
```

Comment:
It would be nicer to not have this use so many lines. [You can easily disable black for a few lines](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#code-style).

Reply:
This is very useful! Thanks.